### ISSUE TITLE
Remove some interdependencies in twisted.mail

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8785]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8785|
|Type|enhancement|
|Created|2016-08-25 13:16:52Z|
|Branch|https://github.com/twisted/twisted/tree/8785-tmail-interdeps|

twisted.mail.smtp imports imap4; that's kinda silly. The things it shares should be in a common module that they both import.

<details><summary>Searchable metadata</summary>

```
trac-id__8785 8785
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__8785_tmail_interdeps 8785-tmail-interdeps
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__None None
time__1472131012012685 1472131012012685
changetime__1473882788002865 1473882788002865
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc set owner to @hawkowl|
|-|-|

https://github.com/twisted/twisted/pull/524#issuecomment-242632927

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/524#issuecomment-245768887

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 32c0fdaafd75cd32d8588095194d3bb8bafd7eae
```
#!CommitTicketReference repository="" revision="32c0fdaafd75cd32d8588095194d3bb8bafd7eae"
Merge 8785-tmail-interdeps: Remove some more twisted.mail interdependencies

Author: hawkowl
Reviewer: rodrigc
Fixes: #8785
```
